src/colorizer/state/slices/dataset_slice.ts

```diff
@@ -0,0 +1,40 @@
+import { StateCreator } from "zustand";
+
+import { BackdropSlice } from "./backdrop_slice";
+import { CollectionSlice } from "./collection_slice";
+
+import Dataset from "../../Dataset";
+
+type DatasetSliceState = {
+  datasetKey: string | null;
+  dataset: Dataset | null;
+};
+
+type DatasetSliceActions = {
+  setDataset: (key: string, dataset: Dataset) => void;
+  clearDataset: () => void;
+};
+
+export type DatasetSlice = DatasetSliceState & DatasetSliceActions;
+
+export const createDatasetSlice: StateCreator<CollectionSlice & DatasetSlice & BackdropSlice, [], [], DatasetSlice> = (
+  set,
+  get
+) => ({
+  datasetKey: null,
+  dataset: null,
+
+  setDataset: (key: string, dataset: Dataset) => {
+    ///// Validate dataset-dependent state values /////
+
+    // Switch to new dataset's default backdrop if the current one is not in the new dataset.
+    let backdropKey = get().backdropKey;
+    if (backdropKey === null || !dataset.hasBackdrop(backdropKey)) {
+      backdropKey = dataset.getDefaultBackdropKey();
+    }
```

Comment:
So this is reaching across slices? I remember this being a huge problem for redux, nice that it's easy here

Reply:
Yup, this is the DatasetSlice accessing values defined by BackdropSlice.